test/integration/node-specific/auto_connect.test.ts

```diff
@@ -0,0 +1,769 @@
+    });
+  });
+
+  context(`class AggregationCursor`, () => {
+    const pipeline = [{ $match: { _id: { $type: BSONType.objectId } } }];
+
+    it(`explain()`, async () => {
+      const agg = client.db().collection('test').aggregate(pipeline);
+      await agg.explain().catch(() => null);
+      expect(client).to.have.property('topology').that.is.instanceOf(Topology);
+    });
+
+    it(`close()`, async () => {
+      const agg = client.db().collection('test').aggregate(pipeline);
+      await agg.close().catch(error => {
+        expect.fail('cursor.close should work without connecting: ' + error.message);
+      });
+    });
+
+    it(`forEach()`, async () => {
+      const agg = client.db().collection('test').aggregate(pipeline);
+      await agg.forEach(item => {
+        expect(item).to.be.a('object');
+      });
+      expect(client).to.have.property('topology').that.is.instanceOf(Topology);
+    });
+
+    it(`hasNext()`, async () => {
+      const agg = client.db().collection('test').aggregate(pipeline);
+      await agg.hasNext();
+      expect(client).to.have.property('topology').that.is.instanceOf(Topology);
+    });
+
+    it(`next()`, async () => {
+      const agg = client.db().collection('test').aggregate(pipeline);
+      await agg.next();
+      expect(client).to.have.property('topology').that.is.instanceOf(Topology);
+    });
+
+    it(`toArray()`, async () => {
+      const agg = client.db().collection('test').aggregate(pipeline);
+      await agg.toArray();
+      expect(client).to.have.property('topology').that.is.instanceOf(Topology);
+    });
+
+    it(`tryNext()`, async () => {
+      const agg = client.db().collection('test').aggregate(pipeline);
+      await agg.tryNext();
+      expect(client).to.have.property('topology').that.is.instanceOf(Topology);
+    });
+
+    it(`stream()`, async () => {
+      const agg = client.db().collection('test').aggregate(pipeline);
+      const stream = agg.stream();
+      await once(stream, 'readable');
+      await stream.read();
+      stream.destroy();
+      expect(client).to.have.property('topology').that.is.instanceOf(Topology);
+    });
+  });
+
+  context(`class OrderedBulkOperation`, () => {
+    it.skip(`execute()`, async () => {
+      const bulk = client.db().collection('test').initializeOrderedBulkOp();
+      bulk.find({ a: 1 });
+      await bulk.execute();
+      expect(client).to.have.property('topology').that.is.instanceOf(Topology);
+    }).skipReason = 'TODO(NODE-4263): legacy bulk operations should auto connect';
+  });
+
+  context(`class UnorderedBulkOperation`, () => {
+    it.skip(`execute()`, async () => {
+      const bulk = client.db().collection('test').initializeUnorderedBulkOp();
+      bulk.find({ a: 1 });
+      await bulk.execute();
+      expect(client).to.have.property('topology').that.is.instanceOf(Topology);
+    }).skipReason = 'TODO(NODE-4263): legacy bulk operations should auto connect';
+  });
```

Comment:
Let me unskip these now and we can assert the opposite that they DO throw if run first